.github/workflows/matrix-deploy-azure-function.yml

```diff
@@ -0,0 +1,76 @@
+name: Multi-Stage Build & Deploy (Azure Function)
+
+on:
+  workflow_dispatch:
+  push:
+    branches: ["main", "development"]
+    paths:
+      - "src/Dfe.PlanTech.AzureFunctions/**"
+      - ".github/workflows/matrix-deploy-azure-function.yml"
+      - ".github/workflows/build-azure-function.yml"
+      - ".github/workflows/deploy-azure-function.yml"
+
+jobs:
+  set-env:
+    runs-on: ubuntu-22.04
+    name: Set Environment (Matrix Deploy Azure Function)
+    outputs:
+      branch: ${{ steps.var.outputs.branch }}
+      artifact: ${{ steps.var.outputs.artifact }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+      - id: var
+        run: |
+          GIT_BRANCH=${GITHUB_BASE_REF:-${GITHUB_REF#refs/heads/}}
+          ARTIFACT=Dfe-PlanTech-AzureFunction
+          echo "branch=$GIT_BRANCH" >> $GITHUB_OUTPUT
+          echo "artifact=$ARTIFACT" >> $GITHUB_OUTPUT
+
+  build-azure-function:
+    needs: set-env
+    name: Build Azure Function
+    uses: ./.github/workflows/build-azure-function.yml
+    with:
+      artifact: ${{ needs.set-env.outputs.artifact }}
+
+  deploy-azure-function-for-dev-test:
+    needs: [set-env, build-azure-function]
+    name: Deploy Azure Function
+    strategy:
+      fail-fast: true
+      max-parallel: 2
```

Comment:
Should this not be 1? I believe this means it'd essentially deploy to dev + test at the same time, but we'd want to deploy sequentially.

Reply:
I had it as 1 but figured that it could be done in parallel, since if one were to fail it would fail the other anyways, but I'm happy to change it back to a 1 if needed / I missed something